war/src/main/js/pages/project/builds-card.js

```diff
@@ -0,0 +1,134 @@
+import debounce from "lodash/debounce";
+import behaviorShim from "@/util/behavior-shim";
+
+// Card/item controls
+const buildHistoryPage = document.getElementById("buildHistoryPage");
+const pageSearch = buildHistoryPage.querySelector(".jenkins-search");
+const pageSearchInput = buildHistoryPage.querySelector("input");
+const ajaxUrl = buildHistoryPage.getAttribute("page-ajax");
+const card = document.querySelector("#jenkins-builds");
+const contents = card.querySelector("#jenkins-build-history");
+const container = card.querySelector(".app-builds-container");
+const noBuilds = card.querySelector("#no-builds");
+
+// Pagination controls
+const paginationControls = document.querySelector("#controls");
+const paginationPrevious = document.querySelector("#up");
+const paginationNext = document.querySelector("#down");
+
+// Refresh variables
+let buildRefreshTimeout;
+const updateBuildsRefreshInterval = 5000;
+
+/**
+ * Refresh the 'Builds' card
+ * @param {QueryParameters}  options
+ */
+function load(options = {}) {
+  /** @type {QueryParameters} */
+  const params = Object.assign({}, options, { search: pageSearchInput.value });
+
+  // Avoid fetching if the page isn't active
+  if (document.hidden) {
+    return;
+  }
+
+  fetch(ajaxUrl + toQueryString(params)).then((rsp) => {
+    if (rsp.ok) {
```

Comment:
if its not ok should we log the response and catch network errors and log them as well?

Reply:
<img width="1305" alt="image" src="https://github.com/jenkinsci/jenkins/assets/43062514/5cb166fa-672d-48cf-aa73-547d0d40c64d">

Added a log - this okay?